### ISSUE TITLE
Horizontal `ListView` in nested `ListView`s doesn't work

### ISSUE THREAD

**Issue by QuncCccccc** [flutter/flutter#150236]
### Steps to reproduce

1. Run the sample code
2. Scroll the horizontal list view

### Expected results

Should be able to scroll horizontally.

### Actual results

Doesn't scroll and the scroll bar for the vertical list view is highlighted. This issue happens on both iOS and web but only happens when I'm using trackpad.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [Carousel].

void main() => runApp(const ExampleApp());

class ExampleApp extends StatelessWidget {
  const ExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Carousel Sample'),
        ),
        body: const Example(),
      ),
    );
  }
}

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  State<Example> createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  @override
  Widget build(BuildContext context) {
    return ListView(
      children: <Widget>[
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[50],
        ),
        Center(
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxHeight: 200),
            child: ListView(
              scrollDirection: Axis.horizontal,
              children: List.generate(20, (int index) {
                return ConstrainedBox(constraints: BoxConstraints.tightFor(width: 200), child: Card.outlined(child: SizedBox(width: 200, height: 200, child: Center(child: Text('Item $index')),),),);
              }),
            )
          ),
        ),
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[100],
        ),
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[200],
        ),
      ],
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/36861262/fbb2f7b8-adf2-4dbe-8388-8c35464d4b1e

</details>

Might be related to https://github.com/flutter/flutter/pull/146654#

CC: @HansMuller

**Comment by ANKESH559** [comment]
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';

/// Flutter code sample for [Carousel].

void main() => runApp(const ExampleApp());

class ExampleApp extends StatelessWidget {
  const ExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Carousel Sample'),
        ),
        body: const Example(),
        // Removing the custom scroll behavior
      ),
    );
  }
}

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  State<Example> createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  @override
  Widget build(BuildContext context) {
    return ListView(
      children: <Widget>[
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[50],
        ),
        Center(
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxHeight: 200),
            child: ListView(
              scrollDirection: Axis.horizontal,
              children: List.generate(20, (int index) {
                return ConstrainedBox(
                  constraints: const BoxConstraints.tightFor(width: 200),
                  child: Card(
                    child: SizedBox(
                      width: 200,
                      height: 200,
                      child: Center(child: Text('Item $index')),
                    ),
                  ),
                );
              }),
            ),
          ),
        ),
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[100],
        ),
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[200],
        ),
      ],
    );
  }
}

**Comment by darshankawar** [comment]
@QuncCccccc Hello,
I ran the provided code sample using latest stable version in web and was able to scroll horizontally using trackpad as shown below:


https://github.com/flutter/flutter/assets/67046386/e7f402c4-76a1-4242-910e-68a655b3d7fd

Can you provide the flutter version in which you see the reported behavior ? The video you provided isn't accessible.

**Comment by HansMuller** [comment]
@darshankawar this issue should be checked against the latest master branch which includes https://github.com/flutter/flutter/pull/146654

**Comment by HansMuller** [comment]
I checked the example on macOS and web with the not-yet-landed fixes in https://github.com/flutter/flutter/pull/150275 and found horizontal scrolling on macOS to be a little unpredictable that seems to be due to confusion about the primary scroll controller. If I specified a horizontal scrollbar (and controller) everything seemed to work as expected.

<details open><summary>Code sample with horizontal scrollbar</summary>
import 'package:flutter/material.dart';

/// Flutter code sample for [Carousel].

void main() => runApp(const ExampleApp());

class ExampleApp extends StatelessWidget {
  const ExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Carousel Sample'),
        ),
        body: const Example(),
      ),
    );
  }
}

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  State<Example> createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  ScrollController controller = ScrollController();

  @override
  Widget build(BuildContext context) {
    return ListView(
      children: <Widget>[
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[50],
        ),
        Center(
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxHeight: 200),
            child: Scrollbar(
              controller: controller,
              child: ListView(
                controller: controller,
                scrollDirection: Axis.horizontal,
                children: List.generate(20, (int index) {
                  return ConstrainedBox(constraints: BoxConstraints.tightFor(width: 200), child: Card.outlined(child: SizedBox(width: 200, height: 200, child: Center(child: Text('Item $index')),),),);
                }),
              ),
            ),
          ),
        ),
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[100],
        ),
        Container(
          width: 200,
          height: 200,
          color: Colors.amber[200],
        ),
      ],
    );
  }
}
</details>

**Comment by darshankawar** [comment]
> I checked the example on macOS and web with the not-yet-landed fixes in #150275 and found horizontal scrolling on macOS to be a little unpredictable that seems to be due to confusion about the primary scroll controller.

I also observed this specially on desktop only and not on web with latest master version.

<details>
<summary> master flutter doctor -v </summary>

```

[!] Flutter (Channel master, 3.23.0-13.0.pre.242, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.23.0-13.0.pre.242 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5187cab7bd (2 days ago), 2024-06-14 23:35:34 -0400
    • Engine revision 9779c273aa
    • Dart version 3.5.0 (build 3.5.0-266.0.dev)
    • DevTools version 2.36.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>
